### DISCUSSION TITLE
How to display filtered video frames in ACAP application?

### DISCUSSION THREAD

**Discussion by WilfR** [AxisCommunications/acap3-examples#199] [General] [answered]
## Issue description

I am interested in writing an ACAP application that would operate roughly as follows: Incoming video frames would be "filtered" in some manner (that is, by applying some convolution filter or similar operation, NOT "filtered" in the sense of selecting only frames that meet some criterion) and then the ACAP would display the filtered video frames to the user. To fix ideas, the "acap3-examples/using-opencv" is a useful prototype to demonstrate how to to the filtering operation on incoming video. However that example does not attempt to display the filtered output video, but instead writes a syslog message for each filtered frame. Is it possible to amend the example to illustrate how to have the ACAP display the filtered output video rather than the original input video? If it is not realistic to update the example, can you tell me if what I am trying to do is possible, and if so, point me in the right direction? Thanks!

**Comment by pataxis** [comment]
Hi @WilfR , thanks for reaching out. We'll get back to you when we have had time to look in to your questions.

**Comment by Corallo** [comment, marked as answer]
Hello @WilfR 

You would like your acap to have a web server that expose a page that streams a video with your edited frames.
That should be possible, but might be hard to accomplish, you would have to implement it in C.
We have some examples of web servers, but they are _minimal_, not exactly what you are trying to make [1](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server) [2](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server-using-fastcgi).

If your device[ supports docker containers](https://github.com/AxisCommunications/docker-acap#compatibility), a simpler way to do it would be something similar to the [pose-estimation-with-flask](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/pose-estimator-with-flask) which uses the flask web server to publish a video stream.

**Comment by WilfR** [comment]
Oh thanks very much for the kind reply @Corallo ! This wasn't the kind of architecture that I was thinking about but I understand your suggestion in general and will have a look into the examples for details. Much appreciated!
Cheers,
Wilf
